src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/InlineCompletion/InlineCompletionEndPoint.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Formatting;
+using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer;
+
+internal class InlineCompletionEndpoint : IInlineCompletionHandler
+{
+    // Usually when we need to format code, we utilize the formatting options provided
+    // by the platform. Similar to DefaultCSharpCodeActionResolver we do not have any, so use defaults.
+    private static readonly FormattingOptions s_defaultFormattingOptions = new FormattingOptions()
```

Comment:
There is no FormattingOptionsProvider at this layer.  I believe the best option here is to update the API from the client side to pass along the formatting options.

That way the C# server can respect the same settings as well.  That will need to be done separately though since it requires a client side change.

Reply:
Interesting, is this something we may want to add to the spec or is the expectation that we query for them?

Reply:
I believe it needs to be added to the spec, similar to how they're passed in onautoinsert